core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackServiceMediaPlayer.java

```diff
@@ -328,7 +329,7 @@ protected void smartMarkAsPlayed(Playable media) {
 
         boolean onMediaPlayerError(Object inObj, int what, int extra);
 
-        boolean endPlayback(boolean playNextEpisode, boolean wasSkipped, boolean switchingPlayers);
+        boolean endPlayback(Playable media, boolean playNextEpisode, boolean wasSkipped, boolean switchingPlayers);
```

Comment:
When the media players changed (from local to remote) we'd have situations where, because calls to the local one run asynchronously, the callback would consistently be invoked after the remote player took over, thus checking it for media that wasn't necessarily the same.